src/Microsoft.OData.Edm/Csdl/Parsing/Ast/CsdlTypeDefinition.cs

```diff
@@ -12,16 +12,48 @@ namespace Microsoft.OData.Edm.Csdl.Parsing.Ast
     internal class CsdlTypeDefinition : CsdlNamedElement
     {
         private readonly string underlyingTypeName;
+        private readonly int? maxLength;

```

Comment:
I can't understand it, in my opinion, the facts should go to its underly type of the type definition, right?
For example, maxLength is for the underlying "Edm.String" primitive type.